sdk/resourcemanager/Azure.ResourceManager/README.md

```diff
@@ -52,33 +52,51 @@ Additional documentation for the `Azure.Identity.DefaultAzureCredential` class c
+#### **[Resource].cs**
 
-#### [Resource]Operations
+This represents a full resource client object which contains a **Data** property exposing the details as a **[Resource]Data** type.
+It also has access to all of the operations on that resource without needing to pass in scope parameters such as subscription ID or resource name.  This makes it very convenient to directly execute operations on the result of list calls
+since everything is returned as a full resource client now.
 
-This represents a service client that's scoped to a particular resource. You can directly execute all operations on that client without needing to pass in scope parameters such as subscription ID or resource name.
+```csharp
+ArmClient armClient = new ArmClient(new DefaultAzureCredential());
+string rgName = "myResourceGroup";
+ResourceGroup rg = await armClient.DefaultSubscription.GetResourceGroups().GetAsync(rgName);
+await foreach (VirtualMachine vm in rg.GetVirtualMachines().GetAllAsync())
+{
+    //previously we would have to take the resourceGroupName and the vmName from the vm object
+    //and pass those into the powerOff method as well as we would need to execute that on a separate compute client
+    await vm.StartPowerOff().WaitForCompletionAsync();
+}
+```
+
+#### **[Resource]Data.cs**
 
-#### [Resource]Container
+This represents the model that makes up a given resource. Typically, this is the response data from a service call such as HTTP GET and provides details about the underlying resource. Previously, this was represented by a **Model** class.
+
+#### **[Resource]Container.cs**
 
 This represents the operations you can perform on a collection of resources belonging to a specific parent resource.
-This mainly consists of List or Create operations. For most things, the parent will be a **ResourceGroup**. However, each parent / child relationship is represented this way. For example, a **Subnet** is a child of a **VirtualNetwork** and a **ResourceGroup** is a child of a **Subscription**.
+This object provides most of the logical collection operations.
 
-#### [Resource]
+| Collection Behavior | Container Method |
+|-|-|
+| Iterate/List | GetAll() |
+| Index | Get(string name) |
+| Add | CreateOrUpdate(string name, [Resource]Data data) |
+| Contains | CheckIfExists(string name) |
+| TryGet | GetIfExists(string name) |
 
-This represents a full resource object which contains a **Data** property exposing the details as a **[Resource]Data** type.
-It also has access to all of the operations and like the **[Resource]Operations** object is already scoped
-to a specific resource in Azure.
+For most things, the parent will be a **ResourceGroup**. However, each parent / child relationship is represented this way. For example, a **Subnet** is a child of a **VirtualNetwork** and a **ResourceGroup** is a child of a **Subscription**.
 
```

Comment:
Can we still show a code snippet where all concepts are used? An example would really help to illustrate those abstract definitions